### PR TITLE
feature/INS-184-contacts-tableview Solving conflicts and updating pro…

### PR DESCRIPTION
**Pull request by vierrra**
…ject

Signed-off-by: Renato Vieira <stant@Renato-Vieira.sandbox>

### CODE REVIEW

**Review by andersonfernandes — APPROVED** [review]

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                        bottom:   self.bottomAnchor,
+                        trailing: self.trailingAnchor,
+                        padding:  UIEdgeInsets(top: 2 , left: 4, bottom: 2, right: 4))
+        
+        mainView.layer.applySketchShadow(color: .lightGray, alpha: 0.2, x: 16, y: 1, blur: 0, spread: 0)
+    }
+    
+    fileprivate func configure(name: String, office: String) {
+        nameLabel    = UILabel(frame: CGRect(x: 68, y: 173, width: 210, height: 19))
+        officeLabel  = UILabel(frame: CGRect(x: 68, y: 194, width: 210, height: 14))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Remove extra space from lines 138 and 139.

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                         textWeight:   .regular,
+                         textColor:    .darkGrayStant,
+                         topAnchor:    28,
+                         leftAnchor:   62,
+                         bottomAnchor: 10,
+                         rightAnchor:  82)
+        }
+    }
+    
+    fileprivate func setText(label: UILabel,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+import UIKit
+
+public class ContactsTableViewWithCollapseSearchBar: UIView {
+
+    public var tableView:         ContactsTableView?
+    public var searchView:        DefaultSearchBar?
+    public var emptyMessageLabel: UILabel?
+    
+    weak var tableViewDelegate:   ContactsTableViewDidSelectDelegate?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift:15]: Remove extra spaces.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        
+        self.configureEmptyMessageLabel()
+        self.configureSearchView()
+        self.configureTableView()
+        self.anchorSearchAndTableView()
+        self.setEmptyMessageLabelVisibility()
+    }
+    
+    fileprivate func configureSearchView() {
+        searchView = DefaultSearchBar(frame: CGRect(x: 0, y: 0, width: self.frame.width, height: DefaultSearchBar.searchViewHeight))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Break the parameters. Keep it indented.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    
+    fileprivate func configureTableView() {
+        tableView = ContactsTableView(frame: CGRect(x:      0,
+                                                    y:      DefaultSearchBar.searchViewHeight,
+                                                    width:  self.frame.width,
+                                                    height: self.frame.height))
+        
+        tableView?.register(ContactsCard.self, forCellReuseIdentifier: ContactsCard.identifier())
+        
+        tableView?.configureTableViewWith(constructionList:  constructionSiteList,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Indent constructionSiteList parameter.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                         trailing: self.trailingAnchor,
+                         padding:  UIEdgeInsets(top:    DefaultSearchBar.searchViewHeight,
+                                                left:   0,
+                                                bottom: 0,
+                                                right:  0))
+    }
+    
+    fileprivate func configureEmptyMessageLabel() {
+        emptyMessageLabel?.removeFromSuperview()
+        emptyMessageLabel = UILabel(frame: CGRect(x: 0, y: DefaultSearchBar.searchViewHeight, width: self.frame.width, height: self.frame.height))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Break the parameters. Keep it indented.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        
+        guard let emptyMessageLabel = emptyMessageLabel else { return }
+        emptyMessageLabel.configure(text: emptyMessage, alignment: .left, size: 16, weight: .regular, color: .darkStant)
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Break the parameters. Keep it indented.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import UIKit
+
+extension ContactsTableViewWithCollapseSearchBar: ContactsAndHideSearchTableViewDelegate {
+    public func collapseSearchView() {
+        guard let searchView = searchView, let tableView = tableView else { return }
+        self.removeAllConstraints()
+               
+        UIView.animate(withDuration: 0.2, delay: 0, options: [.transitionCrossDissolve], animations: {
+                searchView.isHidden = true
+                   tableView.removeAllConstraints()
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    
+    public func expandSearchView() {
+        guard let searchView = searchView, let tableView = tableView else { return }
+        self.removeAllConstraints()
+        
+        UIView.animate(withDuration: 0.2, delay: 0, options: [.transitionCrossDissolve], animations: {
+            searchView.isHidden = false
+            tableView.removeAllConstraints()
+            tableView.frame     = CGRect(x:      tableView.frame.minX,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift:52]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        self.present(alert, animated: true)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.view.backgroundColor = .white
+        
+        for i in 0...5 {
+                        
+            constructionSiteList.append(ContactsInformation(name: name + " \(i)", office: office, imageUrl: cardUrl, iconSize: iconSize, mailButton: self.validMailButtons[i], mailImage: UIImage(named: "mail") ?? UIImage(), callButton: self.validPhoneButtons[i], callImage: UIImage(named: "call") ?? UIImage()))
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Break the parameters. Keep it indented.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+            
+            self.setButtonsActionsFor(index: i)
+        }
+        
+        let contactsTableView = ContactsTableViewWithCollapseSearchBar(frame: CGRect(x: 0, y: navigationBarHeight, width: self.view.frame.width, height: self.view.frame.height - navigationBarHeight))
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Break the parameters. Keep it indented.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        
+        self.view.addSubview(contactsTableView)
+        contactsTableView.anchor(top: self.view.topAnchor, leading: self.view.leadingAnchor, bottom: self.view.bottomAnchor, trailing: self.view.trailingAnchor, padding: UIEdgeInsets(top: navigationBarHeight, left: 0, bottom: 0, right: 0))
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Break the parameters. Keep it indented.

**Review by Leandro97 — CHANGES_REQUESTED** [review]

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 1 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift (hunk 2 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 4 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 5 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 6 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 7 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift (hunk 8 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift (hunk 9 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift:52]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 10 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 11 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift (hunk 12 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionContacts/ConstructionContactsViewController.swift]: Done

**Review by Leandro97 — APPROVED** [review]

**Pull request merged**